src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -120,15 +117,16 @@ class AudioPipelineOutput(BaseOutput):
     audios: np.ndarray
 
 
-def is_safetensors_compatible(info) -> bool:
-    filenames = set(sibling.rfilename for sibling in info.siblings)
+def is_safetensors_compatible(filenames, variant=None) -> bool:
     pt_filenames = set(filename for filename in filenames if filename.endswith(".bin"))
     is_safetensors_compatible = any(file.endswith(".safetensors") for file in filenames)
+
     for pt_filename in pt_filenames:
+        _variant = f".{variant}" if (variant is not None and variant in pt_filename) else ""
         prefix, raw = os.path.split(pt_filename)
-        if raw == "pytorch_model.bin":
+        if raw == f"pytorch_model{_variant}.bin":
             # transformers specific
-            sf_filename = os.path.join(prefix, "model.safetensors")
+            sf_filename = os.path.join(prefix, f"model{_variant}.safetensors")
         else:
             sf_filename = pt_filename[: -len(".bin")] + ".safetensors"
```

Comment:
I had a bit of trouble following the is_safetensors_compatible logic. I think I have my head wrapped around it, so I added some docs, a small refactor, and some tests here as a PR into this branch: https://github.com/huggingface/diffusers/pull/2316

Reply:
Will merge into main after variant PR is merged